socket/src/commonTest/kotlin/mqtt/socket/SocketTests.kt

```diff
@@ -178,8 +179,11 @@ class SocketTests {
         } catch (e: Exception) {
             throw e
         } finally {
-            if (port > 0u)
+            if (port > 0u) {
+                delay(200) // gave a delay to ensure surver has time to process the close() request
```

Comment:
I feel like we need to re-think our coroutine solution here. We shouldn't need to add an arbitrary delay to process the close() request

Reply:
close statement is asynchronous, so eventually it will be closed, so I feel there is no need to worry about adding delay for this test.

Reply:
this seems weird. The close method in `NetworkChannel.close()` throws an IOException which means that it can block the thread. I made some adjustments in a socketException2 branch. let me know if that works. I was able to drop the `delay` call here by changing the `NetworkChannel.close()` method to execute on Dispatchers.IO instead

Reply:
we decided to continue to have the delay for the time being.